addons/sale_timesheet/tests/test_performance.py

```diff
@@ -0,0 +1,34 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.sale_timesheet.tests.test_sale_timesheet import TestSaleTimesheet
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestPerformanceTimesheet(TestSaleTimesheet):
+
+    def test_compute_sale_line(self):
+        project = self.env['project.project'].create({
+            'name': 'Billable Project',
+            'allow_billable': True,
+            'partner_id': self.partner_b.id,
+        })
+        tasks = self.env['project.task'].create([{
+            'name': 'Task %s (%s)' % (i, project.name),
+            'project_id': project.id,
+        } for i in range(50)])
+        tasks.sale_line_id = False
+        self.env.invalidate_all()
+        self.env.registry.clear_cache()
+        with self.assertQueryCount(528):
+            tasks._compute_sale_line()
```

Comment:
Sry I forgot to check your test.
It doesn't make lot of sense to call the compute manually (it never happens, and the protection mechanism is not activated).
It is preferable to simulate a real case:
- Create the project with a tons of task (already done)
- invalidate_all (to force all stuff to be recomputed + have a clean field cache), (no need to invalidate the cache of the registry AFAIK)
- Then with the queries count: change the customer of the project + flush_all (to force compute store recomputation):

Reply:
I agree with Rémy. 🙂 

Thanks for your review, btw! 🙂

Reply:
I agree with both of you and will do the change :)

But I just wanted the opinion of @pivi-odoo because he told me that `assertQueryCount()` tests were not really useful, as the query limit is only an upper bound so we cannot easily detect the changes below that limit, and the query number could change quite frequently. They seem to rarely do this kind of tests in perf squad :man_shrugging: